py/server/deephaven/experimental/s3.py

```diff
@@ -23,6 +23,91 @@
     included in the package. This is an opt-out functionality included by default. If not included, importing this
     module will fail to find the java types.
 """
+
+class Credentials(JObjectWrapper):
+    """
+    Credentials object for authenticating with S3 server.
+    """
+    j_object_type = _JCredentials
+
+    def __init__(self, _j_object: jpy.JType):
+        """
+        Initializes the credentials object.
+
+        Args:
+            _j_object (Credentials): the Java credentials object.
+        """
+        self._j_object = _j_object
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self._j_object
+
+    @classmethod
+    def resolving(cls) -> 'Credentials':
```

Comment:
Overall, the docs for this function need work.  They don't give important specifics, they refer to Java, etc.  The user is just left guessing at what to do.